internal/plugins/manifests/samples.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package manifests
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"sigs.k8s.io/kubebuilder/pkg/model/config"
+	"sigs.k8s.io/kubebuilder/pkg/model/file"
+)
+
+var _ file.Template = &kustomization{}
+var _ file.Inserter = &kustomization{}
+
+// kustomization scaffolds or updates the kustomization.yaml in config/samples.
+type kustomization struct {
+	file.TemplateMixin
+
+	// GroupVersionKind is the sample's gvk to add to this scaffold.
+	GroupVersionKind config.GVK
```

Comment:
I'm using a `config.GVK` here since this plugin shouldn't "known" about the resource a phase 1 plugin created or how it was created (namespaced? native API?), just what's in the config file.